File: endpoints/photosCategoria.py
from silence.decorators import endpoint

@endpoint(
    route="/photoscategoria",
    method="GET",
    sql="SELECT * FROM photosCategoria"
)
def get_all():
    pass

###############################################################################

@endpoint(
    route="/photoscategoria/$photoscategoriaId",
    method="GET",
    sql="SELECT * FROM photosCategoria WHERE photosCategoriaId = $photoscategoriaId"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/photoscategoria/photo/$photoId",
    method="GET",
    sql="SELECT nombre, categoriaId FROM Photos NATURAL JOIN photosCategoria NATURAL JOIN Categoria WHERE photoId = $photoId"
)
def get_by_photo():
    pass

###############################################################################

@endpoint(
    route="/photoscategoria/categoria/$catId",
    method="GET",
    sql="SELECT P.*,C.* FROM Photos P NATURAL JOIN photosCategoria NATURAL JOIN Categoria C WHERE categoriaId = $catId AND visibility='public' order by P.date DESC"
)
def get_by_photo():
    pass

###############################################################################

@endpoint(
    route="/photoscategoria/categoria/nombre/$cat",
    method="GET",
    sql="SELECT C.categoriaId FROM Categoria C WHERE C.nombre = $cat"
)
def get_by_photo():
    pass

###############################################################################

@endpoint(
    route="/photoscategoria",
    method="POST",
    sql="INSERT into photosCategoria (photoId, categoriaId) VALUES ($photoId, $categoriaId)",
    description="Creates a new photoscategoria",
    auth_required=True,
)
def create(photoId,categoriaId):
    pass

###############################################################################

@endpoint(
    route="/photoscategoria/$categoriaId",
    method="PUT",
    sql="UPDATE categoria SET nombre = $nombre WHERE categoriaId = $categoriaId",
    description="Updates an existing categoria",
    auth_required=True,
)
def update(nombre):
    pass

###############################################################################

@endpoint(
    route="/photoscategoria/$categoriaId/$photoId",
    method="DELETE",
    sql="DELETE FROM photosCategoria WHERE categoriaId = $categoriaId and photoId=$photoId",
    description="Removes a categoria",
    auth_required=True,
)
def delete():
    pass

File: docs/js/api/photosCategoria.js
"use_strict";
import { BASE_URL, requestOptions } from "./common.js";
const photoscategoriaAPI = {
    getAll: function () {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategoria`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    getById: function (phCategoriaId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategoria/${phCategoriaId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getByPhoto: function (photoId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategoria/photo/${photoId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    getByCategoria: function (categoriaId) {
        return new Promise(function (resolve, reject) {
            axios
                .get(`${BASE_URL}/photoscategoria/categoria/${categoriaId} `, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    create: function(formData) {
        return new Promise(function(resolve, reject) {
            axios
                .post(`${BASE_URL}/photoscategoria`, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    update: function(categoriaId, formData) {
        return new Promise(function(resolve, reject) {
            axios
                .put(`${BASE_URL}/photoscategoria/${categoriaId}`, formData,
                    requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    delete: function(categoriaId,photoId) {
        return new Promise(function(resolve, reject) {
            axios
                .delete(`${BASE_URL}/photoscategoria/${categoriaId}/${photoId} `, requestOptions
                )
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
};
export { photoscategoriaAPI };
